include/picongpu/plugins/Emittance.hpp

```diff
@@ -0,0 +1,1037 @@
+            gSumPos2->deviceToHost( );
+            gSumMomPos->deviceToHost( );
+            gCount_e->deviceToHost( );
+
+            container::HostBuffer<float_64, DIM1> reducedSumMom2( subGrid.getLocalDomain().size.y() );
+            container::HostBuffer<float_64, DIM1> reducedSumPos2( subGrid.getLocalDomain().size.y() );
+            container::HostBuffer<float_64, DIM1> reducedSumMomPos( subGrid.getLocalDomain().size.y() );
+            container::HostBuffer<float_64, DIM1> reducedCount_e( subGrid.getLocalDomain().size.y() );
+            reducedSumMom2.assign( 0.0 );
+            reducedSumPos2.assign( 0.0 );
+            reducedSumMomPos.assign( 0.0 );
+            reducedCount_e.assign( 0.0 );
+
+            // add gSum values from all GPUs using MPI
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedSumMom2,
+                             gSumMom2->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedSumPos2,
+                             gSumPos2->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedSumMomPos,
+                             gSumMomPos->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+            planeReduce->template operator()( /* parameters: dest, source */
+                             reducedCount_e,
+                             gCount_e->getHostBuffer( ).cartBuffer(),
+                             /* the functors return value will be written to dst */
+                             pmacc::algorithm::functor::Add() );
+
+            /** all non-reduce-root processes are done now */
+            if( ! isPlaneReduceRoot )
+                return;
+
+            // gather to file writer
+            container::HostBuffer< float_64, DIM1 > globalSumMom2( subGrid.getGlobalDomain().size.y() );
+            container::HostBuffer< float_64, DIM1 > globalSumPos2( subGrid.getGlobalDomain().size.y() );
+            container::HostBuffer< float_64, DIM1 > globalSumMomPos( subGrid.getGlobalDomain().size.y() );
+            container::HostBuffer< float_64, DIM1 > globalCount_e( subGrid.getGlobalDomain().size.y() );
+
+            // gather y offsets, so we can store our gathered data in the right order
+            int gatherSize = -1;
+            int gatherRank = -1;
+            MPI_CHECK(MPI_Comm_rank( commGather, &gatherRank ));
```

Comment:
The most of the next lines can be removed because `y_off`, `y_siz` and `gatherRank` is not not used anymore